.github/workflows/ci.yml

```diff
@@ -171,11 +102,21 @@ jobs:
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
+      # Restore caches unconditionally. Cache misses are okay.
+      - name: "Fetching cache (git submodules)"
+        uses: actions/cache/restore@58c146cc91c5b9e778e71775dfe9bf1442ad9a12 # v3.2.3
+        with:
+          path: .git/modules
+          key: gitmodules_all_windows_${{ github.sha }}
+          restore-keys: gitmodules_all_windows
+      - name: "Fetching cache (CMake/ccache)"
+        uses: actions/cache/restore@58c146cc91c5b9e778e71775dfe9bf1442ad9a12 # v3.2.3
         with:
-          submodules: false
-      # Update submodules directly using our preferred options. Fetching git
-      # history and source files for large dependencies is particularly slow on
-      # Windows, so we'd rather not rely on the 'checkout' action for this.
+          path: ${{ github.workspace }}/.ccache
+          key: ccache_all_windows_${{ github.sha }}
+          restore-keys: ccache_all_windows
+      # Fetch dependencies.
+      # TODO(scotttodd): Move these into a Docker image.
       - name: "Updating git submodules"
         run: git submodule update --init --jobs 8 --depth 1
```

Comment:
It's possible that the magic 8 here might be different on windows. It's just something I came up with empirically

Reply:
Hmm... even with a very recent cache hit for `.git/modules/`, https://github.com/iree-org/iree/actions/runs/3971047586/jobs/6807491522 took 9 minutes ... most of that to fetch llvm-project :/